replicaset.js

```diff
@@ -0,0 +1,113 @@
+// Based on https://github.com/knex/knex/issues/2253#issuecomment-551610832
+
+const Knex = require('knex');
+const initKnex = require('./initKnex');
+
+exports.createRoundRobinSelectionStrategy = () => {
+    let round = {
+        true: 0,
+        false: 0,
+    };
+    return (pool, isWrite) => {
+        const selected = pool[round[isWrite]];
+        round[isWrite] = (round[isWrite] + 1) % (pool.length);
+        return selected;
+    };
+};
+
+exports.isWriteQuery = (query) => {
+    return ['insert', 'del', 'update'].includes(query.method);
+};
+
+exports.isWriteBuilder = (builder) => {
+    // Enable query context override: knex.select('*').queryContext({ replicaNode: 'write' | 'read' })
```

Comment:
Can this be passed via repostitory method options?
```js
repo.detail({ id : 1 }, { queryContext: { replicaNode: 'write' } })
```

Reply:
I am afraid not, these options are passed to Bookshelf directly, but BS doesn't seem to pass these options directly below to Knex.

For now you can use:
```js
repo.detail({ id : 1 }, { qb => qb.queryContext({ replicaNode: 'write' }) } )
```

Or I can make shortcut for the reposutory to set it via repository option `queryContext` as you suggested.